Servers/UI/OJS.Servers.Ui/ClientApp/src/pages/contest/ContestDetailsPage.tsx

```diff
@@ -0,0 +1,263 @@
+                          text="Results"
+                        />
+                    )
+                    : null}
+                {isUserAdmin
+                    ? (
+                        <>
+                            <LinkButton
+                              type={LinkButtonType.secondary}
+                              to={getAdministrationContestProblemsInternalUrl(contestIdToNumber.toString())}
+                              text="Problems"
+                            />
+                            <LinkButton
+                              type={LinkButtonType.secondary}
+                              to={getAdministrationContestEditInternalUrl(contestIdToNumber.toString())}
+                              text="Edit"
+                            />
+                        </>
+                    )
+                    : null}
+                <LinkButton
+                  id="button-card-compete"
+                  to={getParticipateInContestUrl({
+                      id: contestIdToNumber,
+                      participationType: ContestParticipationType.Compete,
+                  })}
+                  text="Compete"
+                  type={LinkButtonType.secondary}
+                  state={
+                    isOfficial
+                        ? ButtonState.enabled
+                        : ButtonState.disabled
+                }
+                />
+                <LinkButton
+                  id="button-card-practice"
+                  to={getParticipateInContestUrl({
+                      id: contestIdToNumber,
+                      participationType: ContestParticipationType.Practice,
+                  })}
+                  text="Practice"
+                  type={LinkButtonType.secondary}
+                  state={
+                    isOfficial
+                        ? ButtonState.disabled
+                        : ButtonState.enabled
+                }
+                />
+            </div>
+        ),
+        [
+            contestId,
+            contestIdToNumber,
+            getContestResultsUrl,
+            getParticipateInContestUrl,
+            isOfficial,
+            isUserParticipant,
+            isUserAdmin,
+            participationType,
+            getAdministrationContestProblemsInternalUrl,
+            getAdministrationContestEditInternalUrl,
+        ],
+    );
+
+    const renderResources = useCallback(
+        (resources: IProblemResourceType[]) => resources.map((r) => <ProblemResource resource={r} />),
+        [],
+    );
+
+    const renderTask = useCallback(
+        (problem: IContestDetailsProblemType) => {
+            const { resources } = problem;
+            if (isNil(resources)) {
+                return (<div>{problem.name}</div>);
+            }
+
+            return (
+                <div>
+                    <div className={styles.taskSideNavigationItem}>{problem.name}</div>
+                    {renderResources(resources)}
```

Comment:
I think this approach is more readable:
```
   const renderResource = useCallback(
        (resource: IProblemResourceType) => <ProblemResource resource={resource} />,
        [],
    );
```

    const renderTask = useCallback(
        (problem: IContestDetailsProblemType) => {
            const { resources } = problem;
            if (isNil(resources)) {
                return (<div>{problem.name}</div>);
            }

            return (
                <div>
                    <div className={styles.taskSideNavigationItem}>{problem.name}</div>
                    <List
                      values={resources}
                      itemFunc={renderResource}
                      orientation={Orientation.horizontal}
                      wrap
                    />
                </div>
            );
        },
        [ renderResource ],
    );`
Refactor List properties if needed.

Reply:
Done.